src/multiCursor.ts

```diff
@@ -17,3 +17,26 @@ const addCursorFactory =
 
 export const addCursorUp = addCursorFactory('up');
 export const addCursorDown = addCursorFactory('down');
+
+export const addCursorAtEachSelectionLine: Command = (view) => {
+  const cursors = view.state.selection.ranges.reduce((acc, r) => {
+    if (r.empty) {
+      return acc;
+    }
+
+    for (let pos = r.from; pos <= r.to; ) {
+      const line = view.state.doc.lineAt(pos);
+      acc = [...acc, EditorSelection.cursor(line.to > r.to ? r.to : line.to)];
+
+      pos = line.to + 1;
+    }
+
+    return acc;
+  }, [] as Array<SelectionRange>);
+
+  view.dispatch({
+    selection: EditorSelection.create(cursors, 0),
+  });
+
+  return Boolean(cursors.length);
```

Comment:
nit: might as well be explicit here
```suggestion
  return cursors.length > 0;
```

Reply:
same shit lol

Reply:
haha ultra nit but it's clearer imo